cmd/kubelet/app/server.go

```diff
@@ -598,6 +623,87 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.KubeletDeps) (err error) {
 	return nil
 }
 
+func updateTransport(clientConfig *restclient.Config, clientCertificateManager certificate.Manager) error {
+	pool := x509.NewCertPool()
+	if len(clientConfig.CAFile) > 0 {
+		if certs, err := certutil.NewPool(clientConfig.CAFile); err != nil {
+			return fmt.Errorf("unable to load the CA from %q: %v", clientConfig.CAFile, err)
+		} else {
+			pool = certs
+		}
+	}
+	if len(clientConfig.CAData) > 0 {
+		if certs, err := certutil.ParseCertsPEM(clientConfig.CAData); err != nil {
+			return fmt.Errorf("unable to parse the CA from the data in the config file: %v", err)
+		} else {
+			for _, cert := range certs {
+				pool.AddCert(cert)
+			}
+		}
+	}
+	clientConfig.Transport = utilnet.SetTransportDefaults(&http.Transport{
+		TLSClientConfig: &tls.Config{
```

Comment:
The unserializable fields introduced by the cert manager were in an earlier revision of this PR and have been abandoned due to the serialization concerns.